lib/readAccessor.js

```diff
@@ -0,0 +1,112 @@
+    var attributeCount = accessor.count;
+    var componentCount = numberOfComponentsForType(accessor.type);
+    var componentBytes = byteLengthForComponentType(accessor.componentType);
+    var attributeStride = componentCount * componentBytes;
+    var accessorType = accessor.type;
+    var componentType = accessor.componentType;
+    var attributes = [];
+    var currentBufferIndex = accessor.byteOffset + bufferView.byteOffset;
+
+    var readFunction;
+    switch (componentType) {
+        case 5120: // byte
+            readFunction = function(index) {
+                return source.readInt8(index);
+            };
+            break;
+        case 5121: // unsigned byte
+            readFunction = function(index) {
+                return source.readUInt8(index);
+            };
+            break;
+        case 5122: // short
+            readFunction = function(index) {
+                return source.readInt16LE(index);
+            };
+            break;
+        case 5123: // unsigned short
+            readFunction = function(index) {
+                return source.readUInt16LE(index);
+            };
+            break;
+        case 5126: // float
+            readFunction = function(index) {
+                return source.readFloatLE(index);
+            };
+            break;
+    }
+
+    var generateAttributeFunction;
+    var type = '';
+
+    switch (accessorType) {
+        case 'SCALAR':
+            generateAttributeFunction = function(values) {
+                return values[0];
+            };
+            type = 'number';
+            break;
+        case 'VEC2':
+            generateAttributeFunction = Cartesian2.fromArray;
+            type = 'Cartesian2';
+            break;
+        case 'VEC3':
+            generateAttributeFunction = Cartesian3.fromArray;
+            type = 'Cartesian3';
+            break;
+        case 'VEC4':
+            generateAttributeFunction = Cartesian4.fromArray;
+            type = 'Cartesian4';
+            break;
+        case 'MAT2':
+            generateAttributeFunction = Matrix2.fromArray;
+            type = 'Matrix2';
+            break;
+        case 'MAT3':
+            generateAttributeFunction = Matrix3.fromArray;
+            type = 'Matrix3';
+            break;
+        case 'MAT4':
+            generateAttributeFunction = Matrix4.fromArray;
+            type = 'Matrix4';
+            break;
+    }
+
+    // walk over the buffer, reading the attributes
+    for (var i = 0; i < attributeCount; i++) {
+        // get this attribute's data
+        var values = [];
+        var internalBufferIndex = currentBufferIndex;
+        for (var j = 0; j < componentCount; j++) {
+            values.push(readFunction(internalBufferIndex));
+            internalBufferIndex += componentBytes;
+        }
+        attributes.push(generateAttributeFunction(values));
+        currentBufferIndex += attributeStride;
+    }
+    return {
+        "type" : type,
+        "data" : attributes
```

Comment:
Except when creating test json examples, we usually don't put quotes around the key names.